include/geolib/Shape.h

```diff
@@ -56,6 +58,7 @@ class Shape {
     mutable Vector3 bounding_box_min_cache_;    // cached value of the min corner of a bounding box
     mutable Vector3 bounding_box_max_cache_;    // cached value of the max corner of a bounding box
 
+    static double side_operator(Vector3& p_U, Vector3& p_V, Vector3& q_U, Vector3& q_V);
```

Comment:
I wonder if this is the best place to add this function. It is only used here but I'd like a second opinion

Reply:
Please add documentation as it is unclear to me what it does.

Reply:
Does this need to return a double?

Reply:
The way we use it it can also return a boolean. 
However to keep this function consistent with the mathematical operation it represents it should return a double

Reply:
A static declaration in a header file makes no sense at all, since it's not usable without definition, and being static, the definition is never made available to other compilation units.
If it is a local function, keep it in the same compilation units as where it is used. Preferably move the definition above usage so you don't need to declare the function. If it is not a local function, don't declare it as static.